ADAL/src/workplacejoin/mac/ADWorkPlaceJoinUtil.m

```diff
@@ -121,160 +112,80 @@ + (ADRegistrationInformation *)getRegistrationInformation:(id<MSIDRequestContext
+            
+            if ([challengeIssuerName caseInsensitiveCompare:currentIssuerName] == NSOrderedSame)
+            {
+                identityRef = (__bridge_retained SecIdentityRef)[identityDict objectForKey:(__bridge NSString*)kSecValueRef];
+                *issuer = currentIssuer;
+                break;
+            }
+        }
     }
-    *identity = nil;
     
-    if (*certificate)
+    if (identityList)
     {
-        CFRelease(*certificate);
+        CFRelease(identityList);
+        identityList = NULL;
     }
-    *certificate = nil;
-    
-    *issuer = nil;
-}
-
-
-+ (SecCertificateRef)copyWPJCertificateRef:(id<MSIDRequestContext>)context
-                                     error:(ADAuthenticationError * __nullable __autoreleasing * __nullable)error
-{
-    OSStatus status= noErr;
-    SecCertificateRef certRef = NULL;
-    NSData *issuerTag = [self wpjCertIssuerTag];
-    
-    // Set the private key query dictionary.
-    NSDictionary *queryCert = @{ (__bridge id)kSecClass : (__bridge id)kSecClassCertificate,
-                                 (__bridge id)kSecAttrLabel : issuerTag
-                                 };
     
-    // Get the certificate. If the certificate is not found, this is not considered an error.
-    status = SecItemCopyMatching((__bridge CFDictionaryRef)queryCert, (CFTypeRef*)&certRef);
-    if (status == errSecItemNotFound)
-    {
-        return NULL;
-    }
-    
-    CHECK_KEYCHAIN_STATUS(@"Failed to read WPJ certificate.");
-    
-    return certRef;
-    
-_error:
-    return NULL;
-}
-
-+ (NSData *)wpjCertIssuerTag
-{
-    return [NSData dataWithBytes:certificateIdentifier length:strlen((const char *)certificateIdentifier)];
-}
-
-+ (SecKeyRef)copyPrivateKeyRefForIdentifier:(NSString *)identifier
-                                    context:(id<MSIDRequestContext>)context
-                                      error:(ADAuthenticationError* __nullable __autoreleasing * __nullable)error
-{
-    OSStatus status= noErr;
-    SecKeyRef privateKeyReference = NULL;
-    
-    NSData *privateKeyTag = [NSData dataWithBytes:[identifier UTF8String] length:identifier.length];
-    
-    // Set the private key query dictionary.
-    NSDictionary *privateKeyQuery = @{ (__bridge id)kSecClass : (__bridge id)kSecClassKey,
-                                       (__bridge id)kSecAttrApplicationTag : privateKeyTag,
-                                       (__bridge id)kSecAttrKeyType : (__bridge id)kSecAttrKeyTypeRSA,
-                                       (__bridge id)kSecReturnRef : (__bridge id)kCFBooleanTrue
-                                       };
-    
-    // Get the key.
-    status = SecItemCopyMatching((__bridge CFDictionaryRef)privateKeyQuery, (CFTypeRef*)&privateKeyReference);
-    CHECK_KEYCHAIN_STATUS(@"Failed to read WPJ private key for identifier.");
-    
-    return privateKeyReference;
-    
-_error:
-    return nil;
+    return identityRef; //Caller must call CFRelease
```

Comment:
Question / nit: should we log for nil identityRef here?

Reply:
The caller (getRegistrationInformation) will log for nil identity ref.